### ISSUE TITLE
Generalize `order_with_respect_to` beyond FK fields

### ISSUE THREAD

**Issue by paduszyk** [django-ordered-model/django-ordered-model#299]
### Question

Why the current implementation:

```python
def _wrt_map(self):
    d = {}
    for order_wrt_name in self.get_order_with_respect_to():
        # we know order_wrt_name is a ForeignKey, so use a cheaper _id lookup
        field_path = order_wrt_name + "_id"
        d[order_wrt_name] = get_lookup_value(self, field_path)
    return d
```

which is fixed to FKs, cannot be replaced by:

```python
def _wrt_map(self):
    return {
        order_wrt_name: get_lookup_value(self, order_wrt_name)
        for order_wrt_name in self.get_order_with_respect_to()
    }
```

### Context

In my project, I want to apply ordering of my model w.r.t. to a related objects related to the base model by `ContentType` (usual setup using `content_type` and `object_id` fields). When I override `OrderedModel` class with the method above, everything works as expected.

However, this would be much nicer to have it as a default.

Regards!

**Comment by paduszykam** [comment]
Looks like a duplicate of #298.

**Comment by shuckc** [comment]
If the `order_with_respect_to` field *is* a foreign key, fetching the ID is much cheaper than fetching the related object. Making this change would be a performance regression and we have query count checks in the test suite to make sure this doesn't happen. 
At the moment there is no use of non-foreign keys as OWRT fields in the test suite, however if someone wants to propose the test cases to exercise them then I am open to supporting them again.

**Comment by shuckc** [comment]
Closed as a dupe of #298
